zerver/tests/test_subs.py

```diff
@@ -575,6 +579,7 @@ def test_create_stream_by_admins_only_setting(self):
 
         # Change setting back to default
         do_set_realm_create_stream_by_admins_only(user_profile.realm, False)
+        do_set_realm_waiting_period_threshold(user_profile.realm, initial_threshold)
```

Comment:
I think the "test_create_stream_by_admins_only_setting" test can test the first two cases. For the third case, i can add a new test which checks if a new stream has been created. What do you think ?